File: Ahorn/entities/HorseHitch.jl
module HorseHitch

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/HorseHitch" HorseHitch(x::Integer, y::Integer, destinationId::String="", hitchId::String="", isUnlocked::Bool=false)

const placements = Ahorn.PlacementDict(
    "locked" => Ahorn.EntityPlacement(HorseHitch),
    "unlocked" => Ahorn.EntityPlacement(HorseHitch)
)

function Ahorn.selection(entity::HorseHitch)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::HorseHitch, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end

File: Loenn/entities/horseHitch.lua
local utils = require("utils")
local drawableSprite = require("structs.drawable_sprite")

local horseHitch = {}

horseHitch.name = "MaggyHelper/HorseHitch"
horseHitch.depth = -100
horseHitch.fieldInformation = {}
horseHitch.placements = {
    {
        name = "locked",
        data = {
            hitchId = "",
            destinationId = "",
            isUnlocked = false
        }
    },
    {
        name = "unlocked",
        data = {
            hitchId = "",
            destinationId = "",
            isUnlocked = true
        }
    }
}

function horseHitch.sprite(room, entity)
    local state = entity.isUnlocked and "active" or "inactive"
    local sprite = drawableSprite.fromTexture("objects/horse_hitch/" .. state, entity)
    sprite:setPosition(entity.x, entity.y)
    sprite:setJustification(0.5, 1.0)
    return sprite
end

function horseHitch.selection(room, entity)
    return utils.rectangle(entity.x - 12, entity.y - 40, 24, 40)
end

return horseHitch
